packages/core/src/shared/utilities/zipStream.ts

```diff
@@ -148,14 +160,23 @@ export class ZipStream {
         // We're done streaming all files, so we can close the zip stream
 
         await this._zipWriter.close()
+        const sizeInBytes = this._streamBuffer.size()
+        getLogger().debug('Finished finalizing zipStream with %d bytes of data', sizeInBytes)
         return {
-            sizeInBytes: this._streamBuffer.size(),
+            sizeInBytes,
             hash: this._hasher.digest('base64'),
             streamBuffer: this._streamBuffer,
         }
     }
 
-    public static async unzip(zipBuffer: Buffer): Promise<ZipReaderResult[]> {
+    public async finalizeToFile(targetPath: string, onProgress?: (percentComplete: number) => void) {
+        const result = await this.finalize(onProgress)
+        const contents = result.streamBuffer.getContents() || Buffer.from('')
+        await fs.writeFile(targetPath, contents)
+        return result
+    }
+
+    public static async unzip(zipBuffer: Buffer): Promise<Entry[]> {
         const reader = new ZipReader(new Uint8ArrayReader(new Uint8Array(zipBuffer)))
```

Comment:
off topic: this doesn't actually "stream" the result, it all goes into memory, right? that's something we need to address eventually.

Reply:
I noticed this as well. Also, the current logic in zipUtils dumps this to the file system before uploading whereas `prepareRepoData` uploads the buffer itself. 

If we can centralize the uploading logic (also has two completely separate implementations), this streaming change will be much easier.